### PR TITLE
Gal 198/imd datastore

### PR DESCRIPTION
**Pull request by msandhuNHS**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/environment/dev/resources/dynamodb.tf [hunk 1]
```diff
@@ -133,3 +133,50 @@ resource "aws_dynamodb_table" "gp_practice_table" {
...
+
+  attribute {
+    name = "ImdRank"
+    type = "N"
+  }
+
+  global_secondary_index {
+    name               = "ImdRankImdDecileIndex"
+    hash_key           = "ImdRank"
+    range_key          = "ImdDecile"
```
Comment (LaithM96) [inline terraform/environment/dev/resources/dynamodb.tf:163]: "ImdDecile" is not matched to a defined attribute. That is, you're trying to use "ImdDecile" as a range_key in the aws_dynamodb_table "imd_table", but you didn't define it as an attribute.
You'll need to define "ImdDecile" as an attribute with a specific type (S, N, or B) in your resource block.

you may also need to Re-initialize your Terraform project with terraform init and refresh the state file with terraform refresh to update the status of your infrastructure.

**Review by LaithM96 — CHANGES_REQUESTED** [review]
please read the comments and make the changes to fix the checks.

File: terraform/environment/dev/resources/dynamodb.tf (hunk 1 above)
Comment (msandhuNHS) [reply terraform/environment/dev/resources/dynamodb.tf:168]: All done.

**Review by zainm97 — CHANGES_REQUESTED** [review]

File: terraform/environment/dev/resources/dynamodb.tf [hunk 2]
```diff
@@ -133,3 +133,50 @@ resource "aws_dynamodb_table" "gp_practice_table" {
...
+    type = "S"
+  }
+
+  attribute {
+    name = "ImdRank"
+    type = "N"
+  }
+
+  global_secondary_index {
+    name               = "ImdRankImdDecileIndex"
```
[the remark this answers is not in the document]
Comment (msandhuNHS) [reply terraform/environment/dev/resources/dynamodb.tf:156]: Resolved.

[the remark this answers is not in the document]
Comment (zainm97) [reply terraform/environment/dev/resources/dynamodb.tf:156]: nice

**Review by zainm97 — APPROVED** [review]

**Review by zainm97 — APPROVED** [review]
Looks good

**Review by LaithM96 — APPROVED** [review]
Very good work, happy to approve.

**Pull request merged**